lib/class-wp-theme-json-gutenberg.php

```diff
@@ -871,6 +871,10 @@ protected static function sanitize( $input, $valid_block_names, $valid_element_n
 		$schema['settings']['blocks']                     = $schema_settings_blocks;
 		$schema['settings']['typography']['fontFamilies'] = static::schema_in_root_and_per_origin( static::FONT_FAMILY_SCHEMA );
 
+		if ( ! wp_theme_has_theme_json() ) {
+			$schema['settings']['shadow'] = null;
+		}
```

Comment:
Without it `shadow` setting override in `lib/class-wp-theme-json-resolver-gutenberg.php` is removed by the `sanitize` call.

Reply:
I think it'd be better if we can to avoid overriding a property that's already set by `static::VALID_SETTINGS`. I have an idea, I'll put it in the main review comment 🙂